elks/arch/i86/drivers/block/bioshd.c

```diff
@@ -332,15 +332,42 @@ static int bioshd_open(struct inode *inode, struct file *filp)
     return 0;
 }
 
+void bioshd_park(struct drive_infot *drivep) {
+    unsigned int cyls = drivep->cylinders;
+    if (cyls > 1024 || drivep->fdtype != -1) return; /* int 13 AH=0xC supports only 10 bits for cyl */
+    unsigned char bios_drive = bios_drive_map[drivep - drive_info];
+    unsigned short cyl_sec = ((cyls & 0xFF) << 8) | ((cyls & 0x300) >> 2) | 0x1; /* 0x1 = sector */
+    asm (
+        "movb $0xC,%%ah\n\t"
+        "movw %0,%%cx\n\t"
+        "movb $0x0,%%dh\n\t"
+        "movb %1,%%dl\n\t"
+        "int  $0x13\n\t"
+        : /* No outputs */
+        : "r" (cyl_sec), "r" (bios_drive)
+        : "ax", "cx", "dx"
+    );
```

Comment:
Let's use the existing `call_bios` function to execute an INT 13h, instead of adding ASM code within C. See elks/arch/i86/drivers/block/bios.c for examples. The call_bios function itself lives in elks/arch/i86/lib/bios13.S for reference.